File: front/src/dto/add_xml/updateWkTblAddByXmlTableListCapsuleDto.ts
import FrameworkCapsuleDto from "../frameworkCapsuleDto";
import type WkTblMasterAllByXmlInterface from "../../entity/wkTblMasterAllByXmlEntity";

export default interface UpdateWkTblAddByXmlTableListCapsuleInterface {

}

export default class UpdateWkTblAddByXmlTableListCapsuleDto extends FrameworkCapsuleDto
    implements UpdateWkTblAddByXmlTableListCapsuleInterface {

    /** 編集対象リスト */
    listWkTblByXml: WkTblMasterAllByXmlInterface[];

    constructor() {
        super();
        this.listWkTblByXml = [];
    }
}

File: back/src/main/java/mitei/mitei/political/balancesheet/manage/kanrensha/dto/add_xml/UpdateWkTblAddByXmlTableListCapsuleDto.java
package mitei.mitei.political.balancesheet.manage.kanrensha.dto.add_xml;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

import mitei.mitei.political.balancesheet.manage.kanrensha.dto.FrameworkCapsuleDto;
import mitei.mitei.political.balancesheet.manage.kanrensha.entity.WkTblMasterAllByXmlEntity;

/**
 * XML一括登録最小マスタ編集リストワークテーブル更新CapsuleDto
 */
public class UpdateWkTblAddByXmlTableListCapsuleDto extends FrameworkCapsuleDto // NOPMD DataClass
        implements Serializable {

    /** Serialize id */
    private static final long serialVersionUID = 1L;

    /** 編集対象リスト */
    private List<WkTblMasterAllByXmlEntity> listWkTblByXml = new ArrayList<>();

    /**
     * 編集対象リストを取得する
     *
     * @return 編集対象リスト
     */
    public List<WkTblMasterAllByXmlEntity> getListWkTblByXml() {
        return listWkTblByXml;
    }

    /**
     * 編集対象リストを設定する
     *
     * @param listWkTblByXml 編集対象リスト
     */
    public void setListWkTblByXml(final List<WkTblMasterAllByXmlEntity> listWkTblByXml) {
        this.listWkTblByXml = listWkTblByXml;
    }

}
